pkg/datastore/headers.go

```diff
@@ -0,0 +1,433 @@
+package datastore
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"regexp"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/blocknative/dreamboat/pkg/structs"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/flashbots/go-boost-utils/types"
+	ds "github.com/ipfs/go-datastore"
+)
+
+const (
+	HeaderPrefix        = "header-"
+	HeaderContentPrefix = "hc/"
+
+	InMemorySlotLag        = 200
+	InMemorySlotPurgeCheck = time.Duration(time.Second * 10)
+	InMemorySlotTimeLag    = time.Duration(time.Minute * 1)
+)
+
+var (
+	HeaderPrefixBytes = []byte("header-")
+)
+
+func HeaderKeyContent(slot structs.Slot, blockHash string) ds.Key {
+	return ds.NewKey(fmt.Sprintf("hc/%d/%s", slot, blockHash))
+}
+
+func HeaderMaxNewKey(slot structs.Slot) ds.Key {
+	return ds.NewKey(fmt.Sprintf("hm/%d", slot))
+}
+
+func HeaderKey(slot uint64) ds.Key {
+	return ds.NewKey(fmt.Sprintf("%s%d", HeaderPrefix, slot))
+}
+
+// OLD Entries
+func HeaderMaxProfitKey(slot structs.Slot) ds.Key {
+	return ds.NewKey(fmt.Sprintf("header/max-profit/%d", slot))
+}
+
+type StoredIndex struct {
+	Index                []IndexEl
+	MaxProfit            IndexEl
+	SubmissionsByPubKeys map[[48]byte]IndexEl
+}
+
+func NewStoreIndex() StoredIndex {
+	return StoredIndex{
+		SubmissionsByPubKeys: make(map[[48]byte]IndexEl),
+	}
+}
+
+type IndexEl struct {
```

Comment:
What does `El` stand for?

Reply:
Element? It's meant to be shorter

Reply:
It's the element of the index, so if we renamed is as Stored it should be rather StoredIndexElement.  but it's quite long

Reply:
Okay I see, what about `IndexMeta`? Since it's the metadata about the index, and `El` is not intuitively related to element

Reply:
Added as IndexMeta